parlai/core/torch_agent.py

```diff
@@ -1810,14 +1847,19 @@ def load_state_dict(self, state_dict):
         except RuntimeError as msg:
             msg_ = str(msg)
             if 'size mismatch' in msg_ and 'embedding' in msg_:
-                raise RuntimeError(
-                    f'{msg_}\n'
-                    '-----------------\n'
-                    'Could not load the model due to a size mismatch in the '
-                    'embeddings. A common reason for this is trying to load '
-                    'a model trained with fp16 but loaded without fp16. Try '
-                    'adding --fp16 true or --force-fp16-tokens true.'
-                )
+                if hasattr(self, 'special_toks') and len(self.special_toks) > 0:
+                    state_dict = self._resize_token_embeddings(state_dict, msg_)
+                    self.model.load_state_dict(state_dict)
+                    self.resized = True  # make note that we resized here
```

Comment:
could we change this to `self.resized_embeddings` for clarity? since resized could mean a number of things...